sentry_streams/sentry_streams/adapters/arroyo/reduce.py

```diff
@@ -66,6 +75,64 @@ def accumulator(self, result: Any, value: BaseValue[RoutedValue]) -> RoutedValue
         return routed
 
 
+class MetricsAccumulator(Accumulator[Message[Any], Any]):
+
+    def __init__(self, acc: Callable[[], Accumulator[Any, Any]], name: str) -> None:
+        self.acc = acc()
+        self.start_time: float | None = None
+        self.metrics = get_metrics()
+        self.tags = {"step": name}
+
+    def add(self, value: Message[Any]) -> Self:
+        if self.start_time is None:
+            self.start_time = time.time()
+        self.metrics.increment(Metric.INPUT_MESSAGES, tags=self.tags)
+        size = get_size(value.payload)
+        if size is not None:
+            self.metrics.increment(Metric.INPUT_BYTES, tags=self.tags, value=size)
+
+        self.acc.add(value.payload)
+
+        return self
+
+    def get_value(self) -> Any:
+        result = self.acc.get_value()
+        self.metrics.increment(Metric.OUTPUT_MESSAGES, tags=self.tags)
+        size = get_size(result)
+        if size is not None:
+            self.metrics.increment(Metric.OUTPUT_BYTES, tags=self.tags, value=size)
+
+        duration = time.time() - self.start_time if self.start_time is not None else 0
+        self.metrics.timing(Metric.DURATION, duration, tags=self.tags)
+        return result
+
+    def merge(self, other: Self) -> Self:
+        self.acc.merge(other.acc)
+
+        return self
+
+
+class MetricsReduce(PipelineReduce[MeasurementUnit, InputType, OutputType]):
```

Comment:
Same as above `MetricsReportingReducer` or others? This does not reduce metrics but it is a Reducer that reports metrics